lib/platform.h

```diff
@@ -144,21 +146,19 @@ namespace cppcheck {
                 return "unix64";
             case PlatformFile:
                 return "platformFile";
-            default:
-                return "unknown";
             }
         }
 
         long long unsignedCharMax() const {
-            return max_value(char_bit + 1);
+            return max_value(static_cast<int>(char_bit + 1));
```

Comment:
char_bit should be signed also.

Reply:
However, signature of max_value function is type int.